### DISCUSSION TITLE
How calculate fit_cec_sam parametrs?

### DISCUSSION THREAD

**Discussion by JBreidaks** [pvlib/pvlib-python#2015] [Q&A] [answered]
from pvlib.ivtools.sdm import fit_cec_sam

#https://www.jasolar.com/uploadfile/2022/0511/20220511060835815.pdf
#Model "JA Solar JAM72S30-530/MR"
# Panel specifications
celltype = "monoSi"
v_mp = 41.31  # in volts
i_mp = 12.83  # in amperes
v_oc = 49.3   # in volts
i_sc = 13.72  # in amperes
alpha_sc = 0.65   # in %/C (assuming the same as in the previous example)
beta_voc = -0.286  # in %/C (assuming the same as in the previous example)
gamma_pmp = -0.35  # in %/C
cells_in_series = 144
temp_ref = 25  # in degrees Celsius

# Calculate coefficients
coefficients = fit_cec_sam(celltype, v_mp, i_mp, v_oc,
                           i_sc, alpha_sc, beta_voc,
                           gamma_pmp, cells_in_series)
print(coefficients)

I am trying fit coefficients for CEC SAM, but I get error.

6parsolve simulation error
 : Could not solve, sanity check failed (-33): abs((P - Pmp) / Pmp) > 0.015

RuntimeError: Parameter estimation failed

**Comment by cwhanse** [comment]
> alpha_sc = 0.65 # in %/C (assuming the same as in the previous example)
beta_voc = -0.286 # in %/C (assuming the same as in the previous example)

These temperature coefficients need to be in A/C and V/C, respectively. See the docstring [here](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.ivtools.sdm.fit_cec_sam.html).

**Reply by cwhanse** [reply]
What is the "previous example" you mention?

**Reply by JBreidaks** [reply]
perc_alpha_sc = 0.045
perc_beta_voc = -0.275  # in %/C
alpha_sc = (perc_alpha_sc / 100) * i_sc
beta_voc = (perc_beta_voc / 100) * v_oc

Is this transformation correct?

**Reply by cwhanse** [reply, marked as answer]
Yes
